pkg/crinit/aggregated/aggregated.go

```diff
@@ -0,0 +1,412 @@
+		Example: aggregated_example,
+		Run: func(cmd *cobra.Command, args []string) {
+			err := opts.SetName(args)
+			if err != nil {
+				glog.Fatalf("error: %v", err)
+			}
+
+			err = validateOptions(opts)
+			if err != nil {
+				glog.Fatalf("error: %v", err)
+			}
+
+			err = opts.MarshalOptions()
+			if err != nil {
+				glog.Fatalf("error: %v", err)
+			}
+
+			hostConfig, err := util.GetClientConfig(pathOptions, opts.Host, opts.Kubeconfig).ClientConfig()
+			if err != nil {
+				glog.Fatalf("error: %v", err)
+			}
+			hostClientset, err := client.NewForConfig(hostConfig)
+			if err != nil {
+				glog.Fatalf("error: %v", err)
+			}
+			apiServiceClientset, err := apiregclient.NewForConfig(hostConfig)
+			if err != nil {
+				glog.Fatalf("error: %v", err)
+			}
+			err = Run(opts, cmdOut, hostClientset, apiServiceClientset, pathOptions)
+			if err != nil {
+				glog.Fatalf("error: %v", err)
+			}
+		},
+	}
+
+	flags := cmd.Flags()
+	opts.BindCommon(flags, defaultServerImage, defaultEtcdImage)
+	opts.Bind(flags)
+
+	return cmd
+}
+
+// validateOptions ensures that options are valid.
+func validateOptions(opts *aggregatedClusterRegistryOptions) error {
+	opts.ApiServerServiceType = v1.ServiceType(opts.apiServerServiceTypeString)
+	return opts.ValidateCommonOptions()
+}
+
+// Run initializes a cluster registry.
+func Run(opts *aggregatedClusterRegistryOptions, cmdOut io.Writer,
+	hostClientset client.Interface, apiSvcClientset apiregclient.Interface, pathOptions *clientcmd.PathOptions) error {
+
+	// Create Cluster Registry Namespace
+	err := opts.CreateNamespace(cmdOut, hostClientset)
+	if err != nil {
+		return err
+	}
+
+	// Create Cluster Registry Service
+	svc, ips, hostnames, err := opts.CreateService(cmdOut, hostClientset)
+	if err != nil {
+		return err
+	}
+
+	// Create credentials for secret
+	credentials, err := opts.GenerateCredentials(cmdOut, svc.Name, ips, hostnames,
+		false, false)
+	if err != nil {
+		return err
+	}
+
+	// Create the secret containing the credentials.
+	err = opts.CreateAPIServerCredentialsSecret(hostClientset, credentials)
+	if err != nil {
+		return err
+	}
+
+	// Create Persistent Volume Claim
+	var pvc *v1.PersistentVolumeClaim
+	pvc, err = opts.CreatePVC(cmdOut, hostClientset, svc.Name)
+	if err != nil {
+		return err
+	}
+
+	// Create API server
+	err = opts.CreateAPIServer(cmdOut, hostClientset, false, false, ips, pvc)
+	if err != nil {
+		return err
+	}
+
+	// Create a Kubernetes apiservice. The CA cert above should be base 64
+	// encoded, stripped of new lines and used as the spec.caBundle in the
+	// apiservce. This should not be namespaced
+	_, err = createAPIService(cmdOut, apiSvcClientset, opts,
+		util.GetCAKeyPair(credentials).Cert)
+	if err != nil {
+		return err
+	}
+
+	// Create RBAC objects
+	err = createRBACObjects(cmdOut, hostClientset, opts)
+	if err != nil {
+		return err
+	}
+
+	// Update kubeconfig
+	err = opts.UpdateKubeconfig(cmdOut, pathOptions, svc, ips, hostnames,
```

Comment:
Why do you need to do this? Isn't the idea to access the cluster registry via the kube-apiserver?

Reply:
Yes, although I chose to leave this in because hypothetically you may want to talk directly to the `clusterregistry` API server. For example, I still left the `LoadBalancer` API server service type for this same reason. Thoughts?

Reply:
Hmm. IMO it would be better to make this simpler and let people who want to access the cluster registry directly from kubectl do that themselves. I can't think of a use case that necessitates this, and I'd rather make this code simpler to start and complicate it when there's a user need.